crates/dids-v2/src/bearer_did.rs

```diff
@@ -0,0 +1,182 @@
+
+    let did = DID::parse(&portable_did.uri).map_err(|e| format!("Failed to parse DID: {:?}", e))?;
+
+    let bearer_did = BearerDID {
+        did,
+        key_manager,
+        document: portable_did.document.clone(),
+    };
+
+    Ok(bearer_did)
+}
+
+impl<T: KeyManager> BearerDID<T> {
+    pub fn to_portable_did(&self) -> Result<PortableDID, String> {
+        let private_keys = self
+            .key_manager
+            .export_private_keys()
+            .expect("failed to export private keys");
+
+        let portable_did = PortableDID {
+            uri: self.did.uri.clone(),
+            private_keys: private_keys,
+            document: self.document.clone(),
+        };
+
+        Ok(portable_did)
+    }
+
+    pub fn get_signer<'a>(
+        &'a self,
+        selector: Option<VerificationMethodSelector>,
+    ) -> Result<impl Fn(&[u8]) -> Result<Vec<u8>, String> + 'a, String> {
+        let vm = self
+            .document
+            .select_verification_method(selector)
+            .map_err(|err| format!("Failed to select verification method: {}", err))?;
+
+        let key_alias = self.key_manager.alias(&vm.public_key_jwk).unwrap();
+
+        let signer = move |payload: &[u8]| {
+            self.key_manager
+                .sign(&key_alias, payload)
+                .map_err(|err| format!("Failed to sign payload: {}", err))
+        };
+
+        Ok(signer)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::document::VerificationMethod;
+    use crypto::key::KeyType;
+
+    #[test]
+    fn test_from_portable_did() {
+        let document = Document {
+            id: "did:example:123".to_string(),
+            ..Default::default()
+        };
+        let did = DID::parse(&document.id).unwrap();
+        let key_manager = LocalKeyManager::new_in_memory();
+        let _ = key_manager
+            .generate_private_key(KeyType::Secp256k1)
+            .unwrap();
+        let private_keys = key_manager.export_private_keys().unwrap();
+
+        let portable_did = PortableDID {
+            uri: did.clone().uri,
+            private_keys: private_keys.clone(),
+            document: document.clone(),
+        };
+
+        let bearer_did = from_portable_did(portable_did).unwrap();
+
+        assert_eq!(bearer_did.did, did);
+        assert_eq!(bearer_did.document, document);
+        assert_eq!(
+            bearer_did.key_manager.export_private_keys().unwrap(),
+            private_keys
+        );
+    }
+
+    #[test]
+    fn test_to_portable_did() {
+        let document = Document {
+            id: "did:example:123".to_string(),
+            ..Default::default()
+        };
+        let did = DID::parse(&document.id).unwrap();
+        let key_manager = LocalKeyManager::new_in_memory();
+        let _ = key_manager
+            .generate_private_key(KeyType::Secp256k1)
+            .unwrap();
+
+        let bearer_did = BearerDID {
+            document: document.clone(),
```

Comment:
all these clone's everywhere I should add to my list to review typing and how I could improve this